src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentComplexAttributeContentPass.cs

```diff
@@ -56,6 +56,17 @@ private void ProcessAttributes(TagHelperIntermediateNode node)
                         continue;
                     }
                 }
+                else if (node.Children[i] is TagHelperDirectiveAttributeIntermediateNode directiveAttributeNode)
+                {
+                    if (TrySimplifyContent(directiveAttributeNode) && node.TagHelpers.Any(t => t.IsComponentTagHelper()))
```

Comment:
What's this catch?

Reply:
Just copy-paste from the previous condition. Basically blazor doesn't allow mixed C# + HTML content inside attributes. We want to enforce that even for directive attributes now that they are represented as a different IR node.